pkg/machines/getOSList.py

```diff
@@ -44,6 +53,39 @@
         storage = minimumResources.get_nth(0).get_storage()
         if storage != -1:
             osObj['minimumResources']['storage'] = storage
+
+    osObj['profiles'] = []
+    osInstallScripts = os.get_install_script_list()
+    osObj['profiles'].extend(_getInstallScriptProfile(osInstallScripts))
+
+    osObj['unattendedInstallable'] = False
+    installer_scripts = os.get_install_script_list()
+    if installer_scripts.get_length() > 0:
+        osObj['unattendedInstallable'] = True
+
+    osObj['medias'] = {}
+    osMedias = os.get_media_list()
+    for j in range(osMedias.get_length()):
+        media = osMedias.get_nth(j)
+        mediaId = media.get_id()
+
+        osObj['medias'][mediaId] = {}
+        osObj['medias'][mediaId]['unattendedInstallable'] = False
+        osObj['medias'][mediaId]['profiles'] = []
+
+        if (osObj['unattendedInstallable'] and
+           hasattr(media, 'supports_installer_script')):
+            supports = media.supports_installer_script()
+            osObj['medias'][mediaId]['unattendedInstallable'] = supports
+
+            mediaInstallScripts = media.get_install_script_list()
+            osObj['medias'][mediaId]['profiles'].extend(
+                     _getInstallScriptProfile(mediaInstallScripts))
+
+            if (osObj['medias'][mediaId]['unattendedInstallable'] and
```

Comment:
this is equal to:
`if supports and not osObj['medias'][mediaId]['profiles']):`
and also if there is profile for every script (`_getInstallScriptProfile` result has the same length as its input) then it can be:
`if supports and not mediaInstallScripts`

Reply:
I used the first suggestion. Thanks :) I am not sure if there is profile for every script tbh, so I discarded the second suggestion.